f/connectors/alerts/alerts_gcs.py

```diff
@@ -321,17 +336,33 @@ def prepare_alerts_metadata(alerts_metadata, territory_id):
 
     Returns
     -------
-    list of dict
+    prepared_alerts_metadata : list of dict
         A list of dictionaries representing the filtered and processed alerts
         metadata, including additional columns for geolocation, metadata UUID,
         and alert source.
+    alerts_statistics : dict
+        A dictionary containing alert statistics: total alerts, month/year,
+        and description of alerts.
     """
     # Convert CSV bytes to DataFrame
     df = pd.read_csv(StringIO(alerts_metadata))
 
     # Filter DataFrame based on territory_id
     filtered_df = df.loc[df["territory_id"] == territory_id]
 
+    # Group the DataFrame by month and year, and get the last row for each group
+    filtered_df = (
+        filtered_df.loc[df["territory_id"] == territory_id]
+        .groupby(["month", "year"])
+        .last()
+        .reset_index()
+    )
```

Comment:
I think you might avoid the `.reset_index()` by instead passing `as_index=False` in the groupby

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.groupby.html




```suggestion
    filtered_df = (
        filtered_df.loc[df["territory_id"] == territory_id]
        .groupby(["month", "year"], as_index=False)
        .last()
    )
```